Spring-Boot/Spring-Boot-Problem/src/test/java/tw/waterball/judgegirl/springboot/problem/controllers/ProblemControllerTest.java

```diff
@@ -268,5 +263,75 @@ void WhenSaveProblemWithTitle_ProblemShouldBeSavedAndRespondItsId() throws Excep
+        mockMvc.perform(patch("/api/problems/{problemId}/description", savedProblem.getId())
+                .contentType(MediaType.TEXT_PLAIN_VALUE).content(newDescription))
+                .andExpect(status().isOk());
+        Problem queryProblem = mongoTemplate.findById(savedProblem.getId(), Problem.class);
+        assertNotNull(queryProblem);
+        assertEquals(newDescription, queryProblem.getDescription());
+    }
+
+    @Test
+    void GivenOneProblemSavedAndPatchProblemWithPluginMatchTags_WhenQueryById_ShouldHaveNewTags() throws Exception {
+        Problem savedProblem = givenProblemsSaved(1).get(0);
+        JudgePluginTag pluginMatchTag = new JudgePluginTag();
+        pluginMatchTag.setGroup("Judge Girl");
+        pluginMatchTag.setName("Test");
+        pluginMatchTag.setVersion("1.0");
+        pluginMatchTag.setType(JudgePluginTag.Type.OUTPUT_MATCH_POLICY);
+        mockMvc.perform(patch("/api/problems/{problemId}/plugins/match", savedProblem.getId())
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .content(objectMapper.writeValueAsString(pluginMatchTag)))
+                .andExpect(status().isOk());
+        Problem queryProblem = mongoTemplate.findById(savedProblem.getId(), Problem.class);
+        assertNotNull(queryProblem);
+        JudgePluginTag queryPluginMatchTag = queryProblem.getOutputMatchPolicyPluginTag();
+        assertEquals(
+                objectMapper.writeValueAsString(pluginMatchTag),
+                objectMapper.writeValueAsString(queryPluginMatchTag));
+    }
+
+    @Test
+    void GivenOneProblemSavedAndPatchProblemWithPluginFilterTags_WhenQueryById_ShouldHaveNewTags() throws Exception {
+        Problem savedProblem = givenProblemsSaved(1).get(0);
+        Set<JudgePluginTag> pluginFilterTags = new HashSet<>();
+        final int COUNT_FILTER = 10;
+        for (int i = 1; i <= COUNT_FILTER; i++) {
+            JudgePluginTag pluginFilterTag = new JudgePluginTag();
+            pluginFilterTag.setGroup("Judge Girl");
+            pluginFilterTag.setName(String.format("Test %d", i));
+            pluginFilterTag.setVersion(String.format("%d.0", i));
+            pluginFilterTag.setType(JudgePluginTag.Type.FILTER);
+            pluginFilterTags.add(pluginFilterTag);
+        }
+        mockMvc.perform(patch("/api/problems/{problemId}/plugins/filter", savedProblem.getId())
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .content(objectMapper.writeValueAsString(pluginFilterTags)))
+                .andExpect(status().isOk());
+        Problem queryProblem = mongoTemplate.findById(savedProblem.getId(), Problem.class);
+        assertNotNull(queryProblem);
+        Set<JudgePluginTag> queryPluginFilterTags = new HashSet<>(queryProblem.getFilterPluginTags());
+        assertEquals(
+                objectMapper.writeValueAsString(pluginFilterTags),
+                objectMapper.writeValueAsString(queryPluginFilterTags));
+    }
+
```

Comment:
Improve all of your tests by
1. Split **Given**, **When**, **Then** by line-breaks.
For example, one of your test case should be organized by line-breaks as follows:
```java
@Test
    void GivenOneProblemSavedAndPatchProblemWithTitle_WhenQueryById_ShouldHaveNewTitle() throws Exception {
        Problem savedProblem = givenProblemsSaved(1).get(0);
        String newTitle = UUID.randomUUID().toString();

        mockMvc.perform(patch("/api/problems/{problemId}/title", savedProblem.getId())
                .contentType(MediaType.TEXT_PLAIN_VALUE).content(newTitle))
                .andExpect(status().isOk());

        Problem queryProblem = mongoTemplate.findById(savedProblem.getId(), Problem.class);
        assertNotNull(queryProblem);
        assertEquals(newTitle, queryProblem.getTitle());
    }
```
Now we can easily see the **given**, **when**, **then** with a glimpse.

2. Since your patch APIs are consolidated into one patch API, you should extract a `patchProblem(problemId, request)` method, so the test case above is re-written to:

```java
@Test
    void GivenOneProblemSavedAndPatchProblemWithTitle_WhenQueryById_ShouldHaveNewTitle() throws Exception {
        Problem savedProblem = givenProblemsSaved(1).get(0);
        String newTitle = UUID.randomUUID().toString();

        patchProblem(savedProblem.getId(), 
            PatchProblemUseCase.Request.builder().title(newTitle).build())
                .andExpect(status().isOk());

        Problem queryProblem = mongoTemplate.findById(savedProblem.getId(), Problem.class);
        assertNotNull(queryProblem);
        assertEquals(newTitle, queryProblem.getTitle());
    }
```

3. rename your `queryXXX` to `actualXXX` to indicate this is the actual outcome from the program